src/main/java/com/faforever/api/deployment/LegacyFeaturedModDeploymentTask.java

```diff
@@ -198,12 +199,20 @@ private void updateStatus(String message) {
    * @return the list of deployed files
    */
   @SneakyThrows
-  private List<StagedFile> packageDirectories(Path repositoryDirectory, short version, Map<String, Short> fileIds, Path targetFolder) {
+  @SuppressWarnings("unchecked")
+  private List<StagedFile> packageFiles(Path repositoryDirectory, short version, Map<String, Short> fileIds, Path targetFolder) {
     updateStatus("Packaging files");
     try (Stream<Path> stream = Files.list(repositoryDirectory)) {
-      return stream
-        .filter((path) -> Files.isDirectory(path) && !path.getFileName().toString().startsWith("."))
-        .map(path -> packDirectory(path, version, targetFolder, fileIds))
+      return (List<StagedFile>) stream
+        .map(path -> {
+          if (Files.isDirectory(path) && !path.getFileName().toString().startsWith(".")) {
+            return packDirectory(path, version, targetFolder, fileIds);
+          } else if (Files.isRegularFile(path)) {
+            return packFile(path, version, targetFolder, fileIds);
+          } else{
+            return Optional.empty();
+          }
+        })
         .filter(Optional::isPresent)
         .map(Optional::get)
```

Comment:
instead of map you can flatMap other streams.

packDirectory returns an Optional, so we can call .stream()
packFile also returns an Optional, 
Optional.empty can be replaced with stream.empty

This gets rid of 
        .filter(Optional::isPresent)
        .map(Optional::get)

Reply:
Good point.